src/printsiginfo.c

```diff
@@ -165,6 +165,12 @@ print_si_info(struct tcb *tcp, const siginfo_t *sip)
 		case SIGSEGV: case SIGBUS:
 			tprint_struct_next();
 			PRINT_FIELD_PTR(*sip, si_addr);
+#ifdef HAVE_SIGINFO_T_SI_PKEY
+			if (sip->si_signo == SIGSEGV && sip->si_code == SEGV_PKUERR) {
+				tprint_struct_next();
+				PRINT_FIELD_D(*sip, si_pkey);
```

Comment:
Apparently, siginfo_t.si_pkey is not the only field that could be printed in case of SIGSEGV or SIGBUS,
so I'd suggest to separate `case SIGSEG` and `case SIGBUS`, and use a `switch` on `sip->si_code` instead of an `if` statement.